docs/toolhive/guides-k8s/remote-mcp-proxy.mdx

```diff
@@ -0,0 +1,792 @@
+            resource
+          );
+
+        # Allow users from specific email domain
+        - |
+          permit(
+            principal,
+            action == Action::"tools/call",
+            resource
+          )
+          when {
+            principal has email &&
+            principal.email like "*@company.com"
+          };
+
+        # Role-based access control
+        - |
+          permit(
+            principal,
+            action == Action::"tools/call",
+            resource
+          )
+          when {
+            principal has groups &&
+            principal.groups.contains("admin")
+          };
+
+        # Deny destructive operations
+        - |
+          forbid(
+            principal,
+            action == Action::"tools/call",
+            resource
+          )
+          when {
+            resource.tool in ["delete_data", "drop_table"]
+          };
+```
+
+Available principal attributes depend on the claims in your JWT tokens. Common
+claims include:
+
+- `sub` - Subject (user ID)
+- `email` - Email address
+- `name` - Display name
+- `groups` - Group memberships (array)
+- Custom claims from your identity provider
+
+### Tool filtering
+
+Use `MCPToolConfig` to filter or rename tools from the remote server:
+
+```yaml title="analytics-tools.yaml"
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPToolConfig
+metadata:
+  name: analytics-tools-filter
+  namespace: toolhive-system
+spec:
+  # Only expose read-only tools
+  toolsFilter:
+    - query_data
+    - get_report
+    - list_dashboards
+    - export_chart
+
+  # Rename tools for clarity
+  toolsOverride:
+    query_data:
+      name: analytics_query_data
+      description: Query analytics data with custom filters
+    get_report:
+      name: analytics_get_report
+      description: Retrieve a generated analytics report
+---
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPRemoteProxy
+metadata:
+  name: analytics-proxy
+  namespace: toolhive-system
+spec:
+  remoteURL: https://mcp.analytics.example.com
+  # ... other config ...
+  toolConfigRef:
+    name: analytics-tools-filter
+```
+
+See [Customize tools](./customize-tools.mdx) for more information about tool
+filtering and renaming.
+
+### Token exchange
+
+When your organization has federation with the remote service provider, you can
+configure token exchange to convert company tokens into service-specific tokens:
+
+First, create an `MCPExternalAuthConfig`:
+
+```yaml title="external-auth-config.yaml"
+apiVersion: toolhive.stacklok.dev/v1alpha1
+kind: MCPExternalAuthConfig
+metadata:
+  name: analytics-token-exchange
+  namespace: toolhive-system
+spec:
+  type: tokenExchange
+  tokenExchange:
+    tokenURL: https://auth.company.com/protocol/openid-connect/token
+    clientID: token-exchange-client
+    clientSecretRef:
+      name: token-exchange-creds
+      key: client-secret
+    audience: https://mcp.analytics.example.com
+    scope: 'analytics:read analytics:write'
```

Comment:
Wrong field name (Line ~450): Change scope: (singular, string) to scopes: (plural, array of strings).
  Example should be:
  scopes:
    - 'analytics:read'
    - 'analytics:write'